Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM clearlinux:latest
+
+ENV GOPATH /home/gopath
+ENV PATH="/home/gopath/bin:${PATH}"
+RUN mkdir -p /home/gopath/src/github.com/clearlinux/mixer-tools
+COPY . / /home/gopath/src/github.com/clearlinux/mixer-tools/
+RUN swupd bundle-add mixer go-basic c-basic dev-utils os-core-update-dev
+RUN git config --global user.email "travis@example.com"
+RUN git config --global user.name "Travis CI"
+RUN clrtrust generate
+RUN go get -u gopkg.in/alecthomas/gometalinter.v2
+RUN gometalinter.v2 --install
+
+ENTRYPOINT ["/bin/sh", "-c", "cd /home/gopath/src/github.com/clearlinux/mixer-tools && make && make install && make lint && make check && make batcheck"]
```

Comment:
We talked about having these run separately so even if `make lint` fails we can see the output of the other tests. I bet developers don't want to wait 20 minutes every time to see the **next** failure.

Reply:
By `separately` do you mean changing the `&&` to `;`?

Reply:
Some thoughts:

This isn't _wrong_, but normally you'd put something like this into a script that gets copied over at build and run as the single entrypoint command.

Since all of these commands are executed in the same directory, normally you would use `WORKDIR` to configure this, and forego the `cd`. You'd just want to put it at the end before the `ENTRYPOINT`, though I don't think any of the earlier commands would actually fail if run in that directory, so you could in theory set it anywhere in the `Dockerfile`.

Also, you can override `ENTRYPOINT` at container run time with `--entrypoint`, so can also pretty easily call these steps independently with the rest of the `Dockerfile` as-is.

Another option for breaking this into steps would be to have the scripting outside the Dockerfile -- have the container run in the background, then execute each of the steps inside it from outside with `docker exec <container_id> <cmd>`

Reply:
Yeah I thought of that as well, but I wanted this container to be treated as an executable essentially that is just run, and won't be taking arguments. I don't see this implementation happening for the full docker setup when mixer will only run out of a container, but this works nicely for travis testing IMO.